### PR TITLE
General codebase fixes

### PR DESCRIPTION
**Pull request by hrstoyanov**
What does this PR do?
I ran ArcadeDB through Intellij code inspection and got back a lot of issues.

Motivation
Wanted to improve/modernize the code base.

I did not accept all fixes IntelliJ offered. For example, some of them are related to generics and require more investigation.

Checklist
[X] I have run the build using `mvn clean package` command
[X] My unit tests cover both failure and success scenarios

### CODE REVIEW

**Comment by lvca** [comment]
Hi @hrstoyanov, thanks for your PR! I quickly looked at some of the commits contained in the PR and they are legit. I can't merge the entire PR because it contains some commits I prefer not to include, like the first commit that undoes some changes you did in your branch but are not relevant in the `main` codebase.

Also, I see you fixed some issues in the SQL and GraphQL classes, but that code is automatically generated by JavaCC parser every time we modify the SQL/GraphQL grammars, so they would be lost at the very next change to the grammar.

I'm going to check one by one commit and cherry-pick the relevant ones.

File: engine/src/main/java/com/arcadedb/graph/EdgeIteratorFilter.java [hunk 1]
```diff
@@ -80,7 +80,7 @@ public Edge next() {
 
   @Override
   protected void handleCorruption(final Exception e, final RID edge, final RID vertex) {
-    if ((e instanceof RecordNotFoundException || e instanceof SchemaException) &&//
```
Comment (lvca) [inline engine/src/main/java/com/arcadedb/graph/EdgeIteratorFilter.java:83]: Why did you remove this exception catching?

**Comment by hrstoyanov** [comment]
Thanks Luka. Please cherry pick the commits - this is the best approach. I did try to avoid fixing auto-genetated classes, but I missed - please ignore them.

I did not realize how big of PR it is, so it's best to cherry pick and do this in stages.

File: engine/src/main/java/com/arcadedb/query/sql/executor/MultiValue.java [hunk 2]
```diff
@@ -61,8 +61,7 @@ public static boolean isMultiValue(final Object iObject) {
     }
 
     public static boolean isIterable(final Object iObject) {
-        return iObject instanceof Iterable<?> ||
-                iObject instanceof Iterator<?>;
+        return iObject instanceof Iterator<?>;
```
Comment (lvca) [inline engine/src/main/java/com/arcadedb/query/sql/executor/MultiValue.java]: Why removing the check for iterable?

File: engine/src/main/java/com/arcadedb/query/sql/parser/ContainsTextCondition.java [hunk 3]
```diff
@@ -39,11 +39,11 @@ public ContainsTextCondition(SqlParser p, int id) {
   @Override
   public boolean evaluate(Identifiable currentRecord, CommandContext ctx) {
     Object leftValue = left.execute(currentRecord, ctx);
-    if (leftValue == null || !(leftValue instanceof String)) {
```
Comment (lvca) [inline engine/src/main/java/com/arcadedb/query/sql/parser/ContainsTextCondition.java:42]: Checking for null still makes sense to me

**Comment by hrstoyanov** [comment]
A few things I can try once you are done with absorbing  this batch:
- there were a few fixes I could not understand why IntelliJ suggested. I can try them  again.
- I see the code uses SecurityManager which is to be removed from Java soon (jep 411).
- I see usage of Unsafe for comparisons?
- Generics. This could be sizable work.

File: engine/src/main/java/com/arcadedb/database/async/DatabaseAsyncExecutorImpl.java [hunk 4]
```diff
@@ -175,8 +175,7 @@ public DBAsyncStats getStats() {
     stats.queueSize = 0;
 
     if (executorThreads != null)
-      for (int i = 0; i < executorThreads.length; ++i)
```
Comment (lvca) [inline engine/src/main/java/com/arcadedb/database/async/DatabaseAsyncExecutorImpl.java:178]: The reason why in many core parts we're using the old fashioned loop is for performance: https://howtodoinjava.com/java/collections/performance-comparison-looping-a-list/.

File: e2e/pom.xml [hunk 5]
```diff
@@ -24,7 +24,7 @@
     <parent>
         <groupId>com.arcadedb</groupId>
         <artifactId>arcadedb-parent</artifactId>
-        <version>22.1.1-SNAPSHOT</version>
```
Comment (lvca) [inline e2e/pom.xml:27]: Can't cherry-pick this commit because of this line.

File: engine/src/main/java/com/arcadedb/index/lsm/LSMTreeIndexCursor.java [hunk 6]
```diff
@@ -326,8 +326,8 @@ public RID next() {
           // MERGE VALUES
           final RID[] newArray = Arrays.copyOf(currentValues, currentValues.length + tempCurrentValues.length);
 
-          for (int k = currentValues.length; k < newArray.length; ++k)
-            newArray[k] = tempCurrentValues[k - currentValues.length];
+            if (newArray.length - currentValues.length >= 0)
+                System.arraycopy(tempCurrentValues, currentValues.length - currentValues.length, newArray, currentValues.length, newArray.length - currentValues.length);
```
Comment (lvca) [inline engine/src/main/java/com/arcadedb/index/lsm/LSMTreeIndexCursor.java:330]: It looks like the 2nd parameter `currentValues.length - currentValues.length` is wrong. Shouldn't be just `currentValues.length` ?

File: engine/src/main/java/com/arcadedb/query/sql/executor/CartesianProductStep.java [hunk 7]
```diff
@@ -177,7 +177,7 @@ public void addSubPlan(InternalExecutionPlan subPlan) {
 
   @Override
   public String prettyPrint(int depth, int indent) {
-    String result = "";
```
Comment (lvca) [inline engine/src/main/java/com/arcadedb/query/sql/executor/CartesianProductStep.java:180]: This parts are tests or parts that it's not a problem if they are optimized. I prefer keeping the string concatenation that is more readable. I've merged other commits where it might affect performance.

**Review by lvca — COMMENTED** [review]
Thanks @hrstoyanov, your contribution is much appreciated! I've finished my one by one commi review and I've already merged most of them. Thanks! I see you reverted the last commit but there were nice changes. WDYT about getting a rebase from main and see if you can send more commits by the topic of change?

File: engine/src/main/java/com/arcadedb/database/async/DatabaseAsyncExecutorImpl.java (hunk 4 above)
Comment (arcade-player) [reply engine/src/main/java/com/arcadedb/database/async/DatabaseAsyncExecutorImpl.java:178]: @lvca  maybe you missed the conclusion: 

3. Conclusion

Although the simple for-loop provides the best performance, other looping methods provide much better readability.

Also, **we are using the loop with over one million items** in the list which is not a very practical scenario in most of the applications.

**So if there are not millions of items in the list, use the new Java features such as Stream API or enhanced for-loops.**

Comment (lvca) [reply engine/src/main/java/com/arcadedb/database/async/DatabaseAsyncExecutorImpl.java:178]: Even if the loop does not contain millions of items, that loop can be called millions of times. With the classic iterator/iterable approach, objects are created and destroyed, while the for/loop with the counter avoids that. Even if the advantage is low, every bit helps with performance and every small slowdown added up becomes a noticeable slowdown.

File: engine/src/main/java/com/arcadedb/query/sql/executor/MultiValue.java (hunk 2 above)
Comment (hrstoyanov) [reply engine/src/main/java/com/arcadedb/query/sql/executor/MultiValue.java]: You are right, the check must stay.

File: engine/src/main/java/com/arcadedb/query/sql/parser/ContainsTextCondition.java (hunk 3 above)
Comment (hrstoyanov) [reply engine/src/main/java/com/arcadedb/query/sql/parser/ContainsTextCondition.java:42]: Let's see : 

A. ```(x==null || !(x instanceof Y))```

B. ```(!(x instance of Y))```

If x is NULL , then A is TRUE, B is also TRUE because [null is a sub-class of nothing](https://stackoverflow.com/questions/2950319/is-null-check-needed-before-calling-instanceof)

If x is something and is subclass of Y, then both A and B are false

if x is something that is not a subclass of Y, then both A and B are true


So, it turns out A and B are equivalent for all sorts of x  values, hence the proposed simplification.

File: engine/src/main/java/com/arcadedb/graph/EdgeIteratorFilter.java (hunk 1 above)
Comment (hrstoyanov) [reply engine/src/main/java/com/arcadedb/graph/EdgeIteratorFilter.java:83]: @lvca I always start from the latest code in your repo. I will wait for you to finish picking changes for the last 2 PRs, and continue from the latest code base.

File: engine/src/main/java/com/arcadedb/query/sql/parser/ContainsTextCondition.java (hunk 3 above)
Comment (lvca) [reply engine/src/main/java/com/arcadedb/query/sql/parser/ContainsTextCondition.java:42]: You're right, changing it.

File: e2e/pom.xml (hunk 5 above)
Comment (lvca) [reply e2e/pom.xml:27]: Updated it manually

File: engine/src/main/java/com/arcadedb/index/lsm/LSMTreeIndexCursor.java (hunk 6 above)
Comment (hrstoyanov) [reply engine/src/main/java/com/arcadedb/index/lsm/LSMTreeIndexCursor.java:330]: agreed!

Comment (lvca) [reply engine/src/main/java/com/arcadedb/index/lsm/LSMTreeIndexCursor.java:330]: Also, I think the if should be
```
            if (newArray.length - currentValues.length > 0)
```

Comment (lvca) [reply engine/src/main/java/com/arcadedb/index/lsm/LSMTreeIndexCursor.java:330]: Ok, I can change a little bit that block with 
```java
if (i == 0 || currentValues == null)
  currentValues = tempCurrentValues;
else if (tempCurrentValues.length > 0) {
  // MERGE VALUES
  final RID[] newArray = Arrays.copyOf(currentValues, currentValues.length + tempCurrentValues.length);
  System.arraycopy(tempCurrentValues, currentValues.length, newArray, currentValues.length, newArray.length - currentValues.length);
  currentValues = newArray;
}
```
